src/guiguts/maintext.py

```diff
@@ -739,6 +739,99 @@ def find_matches(
                 start += f"+{count_var.get()}c"
         return matches
 
+    def transform_selection(self, fn) -> None:
+        """Transform a text selection by applying a function or method.
+
+        Args:
+            fn: Reference to a function or method
+        """
+        if not (ranges := self.selected_ranges()):
+            return
+        for _range in ranges:
+            start = _range.start.index()
+            end = _range.end.index()
+            # Trap case where line was too short to select any text
+            # (in do_column_select) and so selection-end was adjusted to
+            # start of next line. Adjust it back again.
+            if _range.end.row > _range.start.row and _range.end.col == 0:
+                end += "- 1l lineend"
+            string = self.get(start, end)
+            self.delete(start, end)
+            # apply transform, then insert at start position
+            self.insert(start, fn(string))
+
+    def sentence_case_transformer(self, s) -> str:
+        """Text transformer to convert a string to "Sentence case".
+
+        The transformation is not aware of sentence structure; if the
+        input string consists of multiple sentences, the result will
+        likely not be what was desired. This behavior was ported as-is
+        from Guiguts 1.x, but could potentially be improved, at least
+        for languages structured like English.
+
+        To be more specific: if multiple sentences are selected, the
+        first character of the first sentence will be capitalized, and
+        the subsequent sentences will begin with a lowercase letter.
```

Comment:
If `Sentence case` is applied to a columnar selection, the first letter of each line of the selection is capitalized. -- I have not yet verified that this same behavior is present in GG1 but am noting it (mind you, I don't know _**why**_ someone would want to apply sentence case to a columnar selection, but I test some weird cases just because). Anyway, should that behavior be noted in a comment?

Confirmed, as I expected, that the caveat about sentence case in a selection also applies to proper names (I'd have been surprised if it hadn't. :D 

Seriously, should Sentence case come with a warning that you may get unexpected results (the same is true of Title Case), so it's always a good idea to check the transformation to make sure. The current GG manual does not.

Reply:
I added a comment (in the docstring). I'm not sure what else to do unless we want to note it in the manual? I don't think GG2 has a manual yet.

FWIW, I copied the behavior from GG1, which is why I was OK leaving it this way. If it hasn't been a problem so far, presumably it's OK?

Reply:
I'm assuming it's okay. As far as I know, no GG2 manual yet. I'll leave it to Nigel to decide whether it needs to be documented or not, but I know that I've seen comments about the title case button in the proofreading interfaces, warning users who use it to check and make sure that it got everything correct.

Reply:
No, there's no GG2 manual yet ;)
I'm happy that it does what makes sense, if the user does a column selection. I don't think there is anything else they would expect it to do, and in a real example, rather than our tests, it would do what you want. For example, a table in a text file with sentences in the middle column. 
It could be argued that sentence case as it is in GG1 (and hence in GG2 atm) scarcely warrants taking up an entry in the menu, but let's keep it as-is for now. In the future it could either be enhanced, dropped or kept.